src/hotspot/share/prims/unsafe.cpp

```diff
@@ -815,6 +815,27 @@ UNSAFE_ENTRY(jboolean, Unsafe_ShouldBeInitialized0(JNIEnv *env, jobject unsafe,
 }
 UNSAFE_END
 
+UNSAFE_ENTRY(void, Unsafe_NotifyStrictStaticAccess0(JNIEnv *env, jobject unsafe, jobject clazz,
+  jlong sfoffset, jboolean writing)) {
+assert(clazz != nullptr, "clazz must not be null");
+
+oop mirror = JNIHandles::resolve_non_null(clazz);
+Klass* klass = java_lang_Class::as_Klass(mirror);
+
+if (klass != nullptr && klass->is_instance_klass()) {
+InstanceKlass* ik = InstanceKlass::cast(klass);
+fieldDescriptor fd;
+if (ik->find_local_field_from_offset((int)sfoffset, true, &fd)) {
+// Note: The Unsafe API takes an OFFSET, but the InstanceKlass wants the INDEX.
```

Comment:
Is the index available for MethodHandles::init_field_MemberName? Should we pass it there?